File: src/main/java/com/converter/api/components/hateoas/ConversionLinkFactory.java
package com.converter.api.components.hateoas;

import com.converter.api.controller.ConversionController;
import org.springframework.hateoas.Link;
import org.springframework.hateoas.server.mvc.WebMvcLinkBuilder;
import org.springframework.stereotype.Component;

@Component
public class ConversionLinkFactory {

    public Link getAllMyConversions(){
        return WebMvcLinkBuilder.linkTo(
                WebMvcLinkBuilder
                        .methodOn(ConversionController.class)
                        .getAll(null, null))
                .withRel("getAllMyConversions")
                .withType("get");
    }

    public Link convert(){
        return WebMvcLinkBuilder.linkTo(
                WebMvcLinkBuilder
                        .methodOn(ConversionController.class)
                        .convert(null,null))
                .withRel("newConversion")
                .withType("post");
    }
}

File: src/main/kotlin/com/converter/api/components/hateoas/ConversionLinkFactory.kt
package com.converter.api.components.hateoas

import com.converter.api.controller.ConversionController
import org.springframework.hateoas.Link
import org.springframework.hateoas.server.mvc.WebMvcLinkBuilder
import org.springframework.stereotype.Component

@Component
class ConversionLinkFactory {

    fun getAllMyConversions() = WebMvcLinkBuilder.linkTo(
            WebMvcLinkBuilder
                .methodOn(ConversionController::class.java)
                .getAll())
            .withRel("getAllMyConversions")
            .withType("get")


    fun convert(): Link  =
         WebMvcLinkBuilder.linkTo(
            WebMvcLinkBuilder
                .methodOn(ConversionController::class.java)
                .convert())
            .withRel("newConversion")
            .withType("post")

}